core/services/ocr2/plugins/promwrapper/plugin.go

```diff
@@ -0,0 +1,141 @@
+package promwrapper
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+var _ types.ReportingPlugin = &promPlugin{}
+
+var (
+	promQuery = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_query_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Query() method",
+			Buckets: []float64{}, // TODO: figure out buckets
```

Comment:
Have we given much thought to this? I wonder if it should be bounded by the maxQueryDuration config for OCR?

Reply:
For the SQL query timeouts, I normalized everything as a percentage of timeout to have common buckets 0-100+. Would that be helpful here? https://github.com/smartcontractkit/chainlink/blob/c2e77ed840f7bb57c6fe7cae1653d35314dd9382/core/services/pg/q.go#L24

Reply:
I'm thinking we can have a wide range of generic timing bucket values here. This wrapper will be used by multiple OCR2 plugins that will have varying execution times:

```
buckets []float64             = []float64{
	float64(1 * time.Millisecond),
        float64(5 * time.Millisecond),
        float64(10 * time.Millisecond),
        float64(50 * time.Millisecond),
        float64(100 * time.Millisecond),
        float64(500 * time.Millisecond),
        float64(time.Second),
        float64(5 * time.Second),
        float64(10 * time.Second),
        float64(30 * time.Second),
        float64(time.Minute),
        float64(2 * time.Minute),
        float64(5 * time.Minute),
        float64(10 * time.Minute),
  }
  ```

Reply:
@jmank88 that's an interesting strategy! I wonder if we should push, in addition to the percentage, the raw value? I feel like that will have some value as well.

@jinhoonbang re: the buckets above, that's also an option 👍

Reply:
I had some logic for this here, in which I made them settable: https://github.com/smartcontractkit/chainlink/pull/7605/commits/cffb0e9ac46c7c475927d3b0a1ea25de56cb93da#diff-56375b2deb9fce0bd05be004966c3d65479f9c031569788f56d6a22a3204fe6fR36

Making it based on OCR configs could be tough because it would have to be fetched from on-chain. I think Jin's way is good for the fact that we get raw performance stats, which we may still want to see early on in the product lifecycle.